c/cert/src/rules/DCL30-C/AppropriateStorageDurationsFunctionReturn.ql

```diff
@@ -0,0 +1,50 @@
+/**
+ * @id c/cert/appropriate-storage-durations-function-return
+ * @name DCL30-C: Declare objects with appropriate storage durations
+ * @description When pointers to local variables are returned by a function it can lead to referring
+ *              to objects outside of their lifetime, which is undefined behaviour.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/dcl30-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import codingstandards.cpp.FunctionParameter
+import semmle.code.cpp.dataflow.DataFlow
+
+class Source extends StackVariable {
+  Source() { not this instanceof FunctionParameter }
+}
+
+abstract class Sink extends DataFlow::Node { }
+
+class FunctionSink extends Sink {
+  FunctionSink() {
+    //output parameter
+    exists(FunctionParameter f |
```

Comment:
Again, we don't need `FunctionParameter` here.
```suggestion
    exists(Parameter f |
```

Reply:
agree, addressed!